src/test/java/com/desiato/puresynth/models/UserControllerTest.java

```diff
@@ -43,6 +45,28 @@ public void createUser_ShouldReturnCreatedUser() throws Exception {
                 .andDo(print());
     }
 
+    @Test
+    public void deleteUser_ShouldDeleteUserAndReturnOk() throws Exception {
+        String uniqueEmail = generateUniqueEmail();
+        String newUserJson = String.format("""
+    {
+        "email": "%s",
+        "password": "password123"
+    }
+    """, uniqueEmail);
+
+        MvcResult result = mockMvc.perform(post("/api/user")
```

Comment:
Up to this line, you could have a helper method to create a user.

There are also different approaches. Some people create TestFactory, where there are methods to create objects for testing (and save them in the DB). Sometimes it's acceptable to also bypass the rest call in the `given` block and call directly a service method to simplify code.

I think it would be reasonable to create a user directly through a service call. We will have quite a few tests, where we will need to create a user. This should speed up testing. Of course the previous test should still have a plain API call.

Reply:
service call created